stubparser/src/org/checkerframework/stubparser/ast/visitor/DumpVisitor.java

```diff
@@ -221,9 +221,22 @@ private void printMembers(List<BodyDeclaration> members, Object arg) {
 
     private void printMemberAnnotations(List<AnnotationExpr> annotations, Object arg) {
         if (annotations != null) {
+            for (Iterator<AnnotationExpr> i = annotations.iterator(); i.hasNext();) {
+                AnnotationExpr a = i.next();
+                a.accept(this, arg);
+                if (i.hasNext()) {
+                    printer.print(" ");
+                }
+            }
+            printer.printLn();
+        }
+    }
+
+    private void printReceiverAnnotations(List<AnnotationExpr> annotations, Object arg) {

```

Comment:
But why do you add so much duplication? You now have three for loops instead of one. If the only difference is whether there is a space before or not, this is possible a lot simpler.
Also, some use extended for loops others use an iterator. This confuses readers and makes one look for some underlying difference - which isn't there.